### PR TITLE
aggreagtions by client-id

### PR DESCRIPTION
**Pull request by kjgarza**
related to https://github.com/datacite/search/issues/168

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -252,7 +252,7 @@ def issn
   end
 
   def registrant_id
-    [subj["provider_id"], obj["provider_id"]].compact
+    [subj["publisher"], obj["publisher"]].compact
```
Comment (mfenner) [inline app/models/event.rb]: Why this change? `registrant-id` is intended as an alias for `provider-id`.

File: app/models/event.rb [hunk 2]
```diff
@@ -171,8 +171,8 @@ def self.query_aggregations
     {
       sources: { terms: { field: 'source_id', size: 50, min_doc_count: 1 } },
       prefixes: { terms: { field: 'prefix', size: 50, min_doc_count: 1 } },
-      pairings: { terms: { field: 'subj.provider_id', size: 50, min_doc_count: 1 }, aggs: { citation_types: { terms: { field: 'obj.provider_id', size: 50, min_doc_count: 1 }, aggs: { "tota" => { sum: { field: 'total' }}}}} },
-      clients: { terms: { field: 'subj.provider_id', size: 50, min_doc_count: 1 }, aggs: { year_months: { date_histogram: { field: 'occurred_at', interval: 'month', min_doc_count: 1 }, aggs: { "total_by_year_month" => { sum: { field: 'total' }}}}} },
+      clients: { terms: { field: 'registrant_id', size: 50, min_doc_count: 1 }, aggs: { year_months: { date_histogram: { field: 'occurred_at', interval: 'month', min_doc_count: 1 }, aggs: { "total_by_year_month" => { sum: { field: 'total' }}}}} },
```
Comment (mfenner) [inline app/models/event.rb]: Please don't use the term `clients`, as this has different meaning for Crossref and DataCite. `providers` is already a bad choice.

**Review by mfenner — CHANGES_REQUESTED** [review]
See two comments. Please address them so that pull request can be merged.

File: app/models/event.rb (hunk 1 above)
Comment (kjgarza) [reply app/models/event.rb]: well the controller is already expecting those are publishers. https://github.com/datacite/lagottino/blob/024c1c035e8ce3d993ef58123017a0e270daeccf/app/controllers/events_controller.rb#L187

I can see levriero to it's sending both:
https://github.com/datacite/levriero/blob/106cb9fbd97ca0b6ad92f2863b8bc2892c7e6552/app/models/base.rb#L220

but the controller only is allowing `publisher`.

I can change the whole thing to one, I just want to hurry up and test-probe that I can definitely use this to make the graphs

Comment (mfenner) [reply app/models/event.rb]: The `publisher` property is something else in events (see `levriero` code), we need `provider_id`. But you are right that we need `provider_id` in line 187 of the events_controller.

File: app/models/event.rb (hunk 2 above)
Comment (kjgarza) [reply app/models/event.rb]: fixed in 0d2ed3a

File: app/models/event.rb (hunk 1 above)
Comment (kjgarza) [reply app/models/event.rb]: plus having the actual name is more useful for the graph. i mean just for my use-case

Comment (kjgarza) [reply app/models/event.rb]: so publisher is not the alphanumeric of provider-id ? mmmmhhh.

Comment (mfenner) [reply app/models/event.rb]: You need identifiers for the aggregation. Human-readable names is something you add in the `facets` concern. I think it is not a good idea to aggregate using a free-text field.

Comment (kjgarza) [reply app/models/event.rb]: fixed on e09599a

**Review by mfenner — APPROVED** [review]
